solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedModelStore.java

```diff
@@ -138,6 +140,24 @@ public LTRScoringAlgorithm makeLTRScoringAlgorithm(String json)
     }
     return makeLTRScoringAlgorithm((Map<String,Object>) parsedJson);
   }
+  
+  private void checkFeatureValidity(LTRScoringAlgorithm meta) throws ModelException {
+    final List<Feature> featureList = meta.getFeatures();
+    final String modelName = meta.getName();
+    if (featureList.isEmpty()) {
+      throw new ModelException("no features declared for model "
+          + modelName);
+    }
+
+    final Set<String> featureNames = new HashSet<>();
+    for (final Feature feature : featureList) {
+      final String fname = feature.getName();
+      if (!featureNames.add(fname)) {
+        throw new ModelException("duplicated feature " + fname + " in model "
```

Comment:
One of them has unit test. I'll make the unit test for no features.
https://github.com/bloomberg/lucene-solr/blob/master-ltr-plugin-develop/solr/contrib/ltr/src/test/org/apache/solr/ltr/feature/TestLTRScoringAlgorithm.java#L98-L112